src/credentials/providers/credentials.ts

```diff
@@ -0,0 +1,120 @@
+/*!
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as telemetry from '../../shared/telemetry/telemetry.gen'
+
+const CREDENTIALS_PROVIDER_ID_SEPARATOR = ':'
+
+/**
+ * "Fully-qualified" credentials structure (source + name).
+ */
+export interface CredentialsId {
```

Comment:
renamed `CredentialsProviderId`. This structure is not a "provider", it is just a value. And it doesn't represent a provider... it represents a set of credentials.